libs/components/datetime/src/lib/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -503,11 +527,29 @@ export class SkyDatepickerInputDirective
       return;
     }
 
-    const dateValue = this.getDateValue(value);
+    let dateValue: Date | string;
+    if (value instanceof Date) {
+      dateValue = this.getDateValue(value);
+    } else {
+      const num = Number(value);
+      if (Number.isInteger(num)) {
+        const now = new Date();
+        const shortcutDate = new Date(now.getFullYear(), now.getMonth(), num);
+        const daysInMonth = shortcutDate.getDate();
+        if (num > 0 && num <= daysInMonth) {
+          dateValue = shortcutDate;
+        } else {
+          dateValue = value;
+        }
+      } else {
```

Comment:
If we could bake this logic into a self-describing private method, that would help for readability.

Reply:
I refactored code to try to make the flow easier to understand and have it be more concise.